### ISSUE TITLE
Bug in BlockFluidBase#getFluidHeightForRender()

### ISSUE THREAD

**Issue by oOMitchOo** [MinecraftForge/MinecraftForge#5296]
The method checks at one point `here.getBlock() instanceof BlockLiquid` instead of using the method `isFluid(here)`, which would also return true, if the block is an `instance of IFluidBlock`.
This results in a render bug at the edges of fluid blocks. In my picture you can see an example of different IFluidBlocks, but this also happens between the same IFluidBlock but with different levels.
I would pull request a fix, but I'm waiting on another pull request of mine.
Only checks for vanilla BlockLiquid
![2018-12-23_12 26 28](https://user-images.githubusercontent.com/11067235/50383277-895b8d00-06b0-11e9-918f-1de8dde859ab.png)
Uses isFluid() instead and checks for IFluidBlock too.
![2018-12-23_12 28 21](https://user-images.githubusercontent.com/11067235/50383278-8d87aa80-06b0-11e9-9926-2a43e4c87c77.png)
I tested this with the newest Forge commit for 1.12.x (from 15 days ago).

**Comment by tterrag1098** [comment]
I don't see why an existing PR should be a blocker for making another...

**Comment by oOMitchOo** [comment]
The truth is the `isFluid(here)` check isn't really a fix.
It uses a vanilla BlockLiquid method after the check, which works because Forge's BlockFluidBase also uses `PropertyInteger LEVEL` with a max of 15 like vanilla.
I think `getFluidHeightForRender()` was intended to work differently. I'll look into it, if I have the time. If anyone is more familiar with the `BlockFluidBase` class, be free to contribute.

**Comment by tterrag1098** [comment]
I don't think that assumption is unique, I believe it's done elsewhere too. It might just be something we have to live with until 1.13
